File: services/customers-service/src/main/java/br/com/deltaglobalbank/customers/infrastructure/persistence/adapters/CustomerAuditRepositoryAdapter.java
package br.com.deltaglobalbank.customers.infrastructure.persistence.adapters;

import java.util.List;

import br.com.deltaglobalbank.customers.domain.audit.CustomerAuditEntry;
import br.com.deltaglobalbank.customers.domain.audit.CustomerAuditRepository;
import br.com.deltaglobalbank.customers.infrastructure.persistence.entities.CustomerAuditEntity;
import br.com.deltaglobalbank.customers.infrastructure.persistence.mappers.CustomerAuditMapper;
import br.com.deltaglobalbank.customers.infrastructure.persistence.repositories.JpaCustomerAuditRepository;
import org.springframework.stereotype.Component;

@Component
public class CustomerAuditRepositoryAdapter implements CustomerAuditRepository {

    private final JpaCustomerAuditRepository jpa;

    public CustomerAuditRepositoryAdapter(JpaCustomerAuditRepository jpa) {
        this.jpa = jpa;
    }

    @Override
    public void saveAll(List<CustomerAuditEntry> entries) {
        List<CustomerAuditEntity> entities = entries.stream()
            .map(CustomerAuditMapper::toEntity)
            .toList();
        jpa.saveAll(entities);
    }
}

File: services/customers-service/src/main/kotlin/br/com/deltaglobalbank/customers/infrastructure/persistence/adapters/CustomerAuditRepositoryAdapter.kt
package br.com.deltaglobalbank.customers.infrastructure.persistence.adapters

import br.com.deltaglobalbank.customers.domain.audit.CustomerAuditEntry
import br.com.deltaglobalbank.customers.domain.audit.CustomerAuditRepository
import br.com.deltaglobalbank.customers.infrastructure.persistence.mappers.toEntity
import br.com.deltaglobalbank.customers.infrastructure.persistence.repositories.JpaCustomerAuditRepository
import org.springframework.stereotype.Component

@Component
class CustomerAuditRepositoryAdapter(
    private val jpa: JpaCustomerAuditRepository,
) : CustomerAuditRepository {
    override fun saveAll(entries: List<CustomerAuditEntry>) {
        jpa.saveAll(entries.map { it.toEntity() })
    }
}